engine/src/main/java/io/zeebe/engine/processor/workflow/deployment/model/validation/ZeebeExpressionValidator.java

```diff
@@ -165,6 +109,53 @@ public Builder(final Class<T> elementType) {
     }
   }
 
+  public static final class ExpressionVerification {
+
+    private boolean isNonStatic = false;
+    private boolean isMandatory = true;
+
+    public ExpressionVerification isNonStatic() {
+      isNonStatic = true;
+      return this;
+    }
+
+    public ExpressionVerification isMandatory() {
+      isMandatory = true;
+      return this;
+    }
+
+    public ExpressionVerification isOptional() {
```

Comment:
Hmm, I am at two minds about this. 

First thought is that the chain `isMandatory().isOptional()` is nonsensical. If the defaults are lenient, then you could omit this method and all methods are only to deviate from the defaults.

Second thought is that I like to have methods which reflect the defaults. This allows to make it explicit at the point where you use this class. Then, the counterpart to `isNonStatic()` is missing.

Third thought is maybe we get rid of the defaults and set them explicitly in the constructor and then we have two factory methods to create optional/mandatory `ExpressionVerification` factory methods. Then the API could be used like this:
```
        ZeebeExpressionValidator.verifyThat(ZeebeInput.class)
            .hasValidExpression(
                ZeebeInput::getSource, ExprssionVerification.mandatory().nonStatic())
```